source/docs/reference/graphql-module-definition.md

```diff
@@ -0,0 +1,181 @@
+---
+id: graphql-module-definition
+title: GraphQL module definition
+---
+
+**Front-Commerce’s GraphQL modules** is the mechanism allowing to extend and override any part of the schema defined by other modules.
+
+<blockquote class="info">
+To get started with GraphQL modules, we recommend you to read the [Extend the GraphQL schema](/docs/essentials/extend-the-graphql-schema.html) documentation page.
+</blockquote>
+
+A Front-Commerce GraphQL module has to export an object containing its definition.
+This page contains the API different keys available in a GraphQL module’s definition.
+
+## `namespace`
+
+The simplest (but useless) GraphQL module definition only requires an unique `namespace` key:
+
+```js
+// A minimal Front-Commerce GraphQL module that basically does nothing
+module.exports = {
+  namespace: "Acme/HelloWorld"
+};
+```
+
+This `namespace` can be used by other modules as a dependency, to ensure another module has been registered in the application.
+
+## `modules` (optional)
+
+The `modules` key is useful when creating a
+[meta module](/docs/advanced/graphql/meta-modules.html).
+It allows to declare the submodules to be registered when this module is registered.
+
+It should be an array with a list of submodules to be included in the application.
+
+Example:
+```js
+import Core from "./core";
+import FeatureA from "./feature-a";
+import FeatureB from "./feature-b";
+
+module.exports = {
+  namespace: "Acme/All",
+  modules: [
+    Core,
+    FeatureA,
+    FeatureB
+  ]
+};
+```
+
+<blockquote class="info">
+**Dependency management:** submodules dependencies will be resolved independently.
+
+It means that if a meta module `A` declares `A1` and `A2` in its `modules` key,
+and `A1` depends on module `B` then there is no guarantee that `A2` will be
+initialized _after_ `B`.
+
+You should make dependencies explicit either in the meta module definition or in
+each submodule definition.
+</blockquote>
+
+## `dependencies` (optional)
+
+A list of module namespaces which must have been registered in the application to allow this module to work. Dependencies will be initialized before this module.
+
+Example:
+```js
+module.exports = {
+  namespace: "Acme/HelloWorld",
+  dependencies: ["Acme/Core"],
+  // …
+};
+```
+
+<blockquote class="warning">
+**Note:** we recommend to try to avoid dependencies between your modules. Usually this is a smell that the schema may be implemented differently and modules are too coupled together. A good use case is for instance to depend on a Front-Commerce core module.
```

Comment:
Is this really a smell? Having many is a smell, but there's almost no chance that a module has no dependency if we're thinking in Graph. For instance, you'll need to add your new field to `Product`, `Category` or anything else. It would not make sense most of the time to add it to the root query.

Reply:
There is a lot to say, and I agree that there are valid use cases. Let’s remove it for now and we will see how people use dependencies ;)